torchgeo/datasets/copernicus/cloud_s3.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Copernicus-Bench Cloud-S3 dataset."""
+
+import os
+from collections.abc import Callable, Sequence
+from typing import Literal
+
+import numpy as np
+import rasterio as rio
+import torch
+from torch import Tensor
+
+from ..utils import Path
+from .base import CopernicusBenchBase
+
+
+class CopernicusBenchCloudS3(CopernicusBenchBase):
+    """Copernicus-Bench Cloud-S3 dataset.
+
+    Cloud-S3 is a cloud segmentation dataset with raw images from Sentinel-3 OLCI
+    and labels from the
+    `IdePix <https://step.esa.int/main/snap-supported-plugins/idepix-tool/>`__
+    classification algorithm.
+
+    This dataset has two modes:
+
+    .. list-table:: Multiclass Classification
+       :header-rows: 1
+
+       * - Code
+         - Class
+         - Description
+       * - 255
+         - Invalid
+         - Invalid pixels, should be ignored during training.
+       * - 0
+         - Clear
+         - Land, coastline, or water pixels.
+       * - 1
+         - Cloud-Sure
+         - Fully-opaque clouds with full confidence of their detection.
+       * - 2
+         - Cloud-Ambiguous
+         - Semi-transparent clouds, or clouds where the detection level is uncertain.
+       * - 3
+         - Cloud Shadow
+         - Pixels are affected by a cloud shadow.
+       * - 4
+         - Snow/Ice
+         - Clear snow/ice pixels.
+
+    .. list-table:: Binary Classification
+       :header-rows: 1
+
+       * - Code
+         - Class
+         - Description
+       * - 255
+         - Invalid
+         - Invalid pixels, should be ignored during training.
```

Comment:
This may or may not work with our `task='binary'` added in #2219, need to figure out how to `ignore_index` with `nn.BCEWithLogitsLoss`.